src/datumaro/plugins/validators.py

```diff
@@ -1196,3 +1208,451 @@ def generate_reports(self, stats):
+
+
+class TabularValidator(_TaskValidator):
+    """
+    A specific validator class for tabular dataset.
+    """
+
+    def __init__(
+        self,
+        task_type=TaskType.tabular,
+        few_samples_thr=None,
+        imbalance_ratio_thr=None,
+        far_from_mean_thr=None,
+        dominance_ratio_thr=None,
+        topk_bins=None,
+    ):
+        super().__init__(
+            task_type=task_type,
+            few_samples_thr=few_samples_thr,
+            imbalance_ratio_thr=imbalance_ratio_thr,
+            far_from_mean_thr=far_from_mean_thr,
+            dominance_ratio_thr=dominance_ratio_thr,
+            topk_bins=topk_bins,
+        )
+
+        self.value_template = {"value": deepcopy(self.numerical_stat_template)}
+
+    def _compute_common_statistics(self, dataset):
+        try:
+            stop_words = set(stopwords.words("english"))  # TODO
+        except LookupError:
+            import nltk
+
+            nltk.download("stopwords")
+            stop_words = set(stopwords.words("english"))  # TODO
+
+        empty_label_template = {"count": 0, "items_with_empty_label": []}
+        empty_caption_template = {"count": 0, "items_with_empty_caption": []}
+        redundancies_template = {
+            "stopword": {"count": 0, "items_with_redundancies": []},
+            "url": {"count": 0, "items_with_redundancies": []},
+        }
+
+        stats = {
+            "label_distribution": {
+                "defined_labels": {},
+                "empty_labels": {},
+            },
+            "caption_distribution": {
+                "defined_captions": {},
+                "empty_captions": {},
+                "redundancies": {},
+            },
+        }
+        stats["total_ann_count"] = 0
+        stats["items_missing_annotation"] = []
+        stats["items_broken_annotation"] = []
+
+        label_dist = stats["label_distribution"]
+        defined_label_dist = label_dist["defined_labels"]
+        empty_label_dist = label_dist["empty_labels"]
+
+        caption_dist = stats["caption_distribution"]
+        defined_caption_dist = caption_dist["defined_captions"]
+        empty_caption_dist = caption_dist["empty_captions"]
+        redundancies_in_caption_dist = caption_dist["redundancies"]
+
+        label_categories = dataset.categories().get(AnnotationType.label, LabelCategories())
+        len_categories = len(dataset._tabular_cat_types.items())
+        caption_columns = [
+            cat
+            for cat, type_ in dataset._tabular_cat_types.items()
+            if not isinstance(type_, CategoricalDtype)
+        ]
+        label_columns = [
+            cat
+            for cat, type_ in dataset._tabular_cat_types.items()
+            if isinstance(type_, CategoricalDtype)
+        ]
+        redundancies_in_caption_dist = {
+            cat: deepcopy(redundancies_template)
+            for cat, type_ in dataset._tabular_cat_types.items()
+            if type_ is str
+        }
```

Comment:
`TabularValidator` seems completely new one and no dependency on existing one. Can we make this first step cleaner? Please introduce static data objects rather than using Python dictionary for internal valuables such as
```suggestion
@dataclasses
class TabularValidationStats:
      total_ann_count: int
      ...
      
      def to_dict(self):
           return {
                "total_ann_count": self.total_ann_count,
                ...
           }

        # No dictionary anymore
        # stats = {...}
        stats = TabularValidationStats()
```

Reply:
Fix in [fed5e27](https://github.com/openvinotoolkit/datumaro/pull/1498/commits/fed5e272400bcd3d4cf336035f58571caaffdcc0). If there is any problem, please let me know.